dashboard/app/models/user.rb

```diff
@@ -1093,15 +1093,21 @@ def recent_courses_and_scripts
     end
 
     user_script_data = user_scripts.map do |user_script|
-      script_id = user_script[:script_id]
-      script = Script.get_from_cache(script_id)
-      {
-        name: script[:name],
-        title: data_t_suffix('script.name', script[:name], 'title'),
-        description: data_t_suffix('script.name', script[:name], 'description_short', default: ''),
-        link: script_path(script),
-      }
-    end
+      # Skip this script if we are excluding the primary script and this is the
+      # primary script.
+      if exclude_primary_script && user_script[:script_id] == primary_script[:id]
```

Comment:
should we capture primary_script outside of the loop? i worry a little that primary_script might be hitting the db each time, tho maybe rails is smart enough to avoid.